File: week0/count_words/solution.py
from collections import defaultdict

# returns a dictionary of the following kind:
# {"word" : count (= occurrences in 'arr')}
def count_words(arr):
    counts = {}
# using defaultdict to avoid checking if
# key is in the dictionary 'counts' or not
    counts = defaultdict(lambda: 0)
    for word in arr:
            counts[word] += 1

    return dict(counts)

File: week0/count_words/solution.rb
def count_words(arr)
	hash = Hash.new(0)
	for word in arr
		hash[word] += 1
	end
	print_hash(hash)
	return hash
end

def print_hash(hash)
	for key in hash.keys
		puts key + " : " + hash[key].to_s
	end
end

count_words(ARGV)